lms/djangoapps/grades/api/v1/views.py

```diff
@@ -243,11 +248,12 @@ def _iter_user_grades(self, course_key, course_enrollment_filter=None, related_m
             enrollments_in_course = enrollments_in_course.select_related(*related_models)
 
         paged_enrollments = self.paginate_queryset(enrollments_in_course)
-        users = (enrollment.user for enrollment in paged_enrollments)
-        grades = CourseGradeFactory().iter(users, course_key=course_key)
+        users = [enrollment.user for enrollment in paged_enrollments]
 
-        for user, course_grade, exc in grades:
-            yield user, course_grade, exc
+        with bulk_course_grade_context(course_key, users), bulk_subsection_grade_context(course_key, users):
```

Comment:
The prefetching of Subsection grades isn't needed when just retrieving Course Grades.  Can we update the code so we don't unnecessarily prefetch grades when not needed by other APIs.

Reply:
Ah, thanks for catching that, will fix.